pkg/actuators/machineset/controller.go

```diff
@@ -127,19 +152,39 @@ func isInvalidConfigurationError(err error) bool {
 func (r *Reconciler) reconcile(machineSet *machinev1.MachineSet) (ctrl.Result, error) {
 	providerConfig, err := getproviderConfig(machineSet)
 	if err != nil {
-		return ctrl.Result{}, mapierrors.InvalidMachineConfiguration("failed to get providerConfig: %v", err)
+		return ctrl.Result{}, machineapierrors.InvalidMachineConfiguration("failed to get providerConfig: %v", err)
+	}
+
+	ibmClient, err := r.getIbmClient(machineSet.GetNamespace(), *providerConfig)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	_, err = ibmClient.InstanceGetProfile(providerConfig.Profile)
+
+	if err != nil {
+		klog.Error("Unable to set annotations: unknown profile: %v", providerConfig.Profile)
+		klog.Error("Autoscaling from zero will not work. To fix this, manually populate machine annotations for your instance profile: %v", []string{profileKey})
+
+		// Returning no error to prevent further reconciliation, as user intervention is now required but emit an informational event
+		r.recorder.Eventf(machineSet, corev1.EventTypeWarning, "FailedUpdate", "Failed to set autoscaling from zero annotations, instance profile unknown")
+		return ctrl.Result{}, nil
 	}
-	instanceType := providerConfig.Profile
 
 	if machineSet.Annotations == nil {
 		machineSet.Annotations = make(map[string]string)
 	}
 
-	// TODO: get annotations keys from machine API
-	// machineSet.Annotations[cpuKey] = strconv.FormatInt(instanceType.VCPU, 10)
-	// machineSet.Annotations[memoryKey] = strconv.FormatInt(instanceType.MemoryMb, 10)
-	// machineSet.Annotations[gpuKey] = strconv.FormatInt(instanceType.GPU, 10)
-	machineSet.Annotations[profileKey] = instanceType
+	// Extract CPU and Memory
+	profileStr := strings.Split(providerConfig.Profile, "-")
+	cpuStr := strings.Split(profileStr[1], "x")[0]
+	memStr := strings.Split(profileStr[1], "x")[1]
```

Comment:
Is this formula guaranteed to always be true going forward?

Reply:
Should be - Has not changed in a very long time.

Reply:
Ok, I think it would be good to have a comment that explains the format of the profile names so that it's obvious what we are doing here if we come back to this please